File: 0141-linked-list-cycle/0141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        if (head == null || head.next == null)
        {
            return false;
        }
        ListNode curr = head;
        ListNode curr1 = head;
        while (curr1.next != null && curr1.next.next != null && curr.next != null)
        {
            curr = curr.next;
            curr1 = curr1.next.next;
            if (curr == curr1)
            {
                return true;
            }
        }
        return false;
    }
}

File: 0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def hasCycle(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        if head == None or head.next == None:
            return False
        curr = head
        currfast = head.next
        while currfast and currfast.next and currfast.next.next:
            if curr == currfast:
                return True
            currfast = currfast.next.next
            curr = curr.next
        return False